pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -356,9 +356,20 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+			if containerID, err := getContainerIDFromLegacyLogSymlink(logSymlink); err == nil {
+				if _, err := cgc.client.ContainerStats(containerID); err == nil {
```

Comment:
It seems weird to use `ContainerStats` here, it is to get container statistics data, not the file system `stats`. :)

Even we want something like `InspectContainer`, we should probably use `ContainerStatus` instead.

Container runtime calls are expensive. 10 `ContainerStatus` is more expensive than 1 `ListContainers`, because:
1) `ContainerStats` contains much more unused information than `ListContainers`, which means more marshal/unmarshal;
2) 10 rpc round trip is more expensive than one.

I think it is preferred to run `getKubeletContainers` once and use the result. :)

Reply:
@Random-Liu 
Your comment has been addressed.

Reply:
Sorry, I was talking about `ContainerStatus`. :P

Reply:
> Though we wouldn't find unhealthy symlink in each invocation of evictPodLogsDirectories()

Hm, that is true. I think you are right. Because the function will only be called when the symlink is unhealthy, even we do `ContainerStatus`, the overhead shouldn't be high (and could be even lower). :P 

If so, I think either `ContainerStatus` or `ListContainers` are good to me. Please feel free to change back to your original code, but please use `ContainerStatus` instead of `ContainerStats`. :)

Reply:
This was the method called in the previous iteration:
```
	ContainerStats(containerID string) (*runtimeapi.ContainerStats, error)
```
I don't see ContainerStatus involved.